stacks-common/src/types/chainstate.rs

```diff
@@ -78,6 +78,7 @@ impl TrieHash {
     }
 
     /// Convert to a String that can be used in e.g. sqlite
+    #[allow(clippy::inherent_to_string_shadow_display)]
```

Comment:
This one may actually be informative. Looks like without this function, it would reach `to_hex` from stacks-common/src/util/hash.rs:636, which is essentially the not unrolled version of this function. Maybe we can just add a comment here that the unrolling is why this version is intentionally replacing that version.